.github/workflows/workflow.yaml

```
@@ -35,3 +38,41 @@ jobs:
           nvim --version
           nvim --headless -c ':PlenaryBustedDirectory lua/spec'
 
+  release:
+    name: release
+    if: ${{ github.ref == 'refs/heads/master' }}
+    needs:
+      - check-formatting
+      - tests
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Setup Node.js
+        uses: actions/setup-node@v1
+        with:
+          node-version: 20
+      - name: Release
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: npx semantic-release
+
+  luarocks-upload:
+    runs-on: ubuntu-22.04
+    needs: release
+    steps:
+      - uses: actions/checkout@v3
+      - name: Get Version 
+        run: echo "LUAROCKS_VERSION=$(git fetch --tags; git describe --abbrev=0 --tags)" >> $GITHUB_ENV
+      - name: LuaRocks Upload
```

required to build the `tree-sitter-go` dependency.

Note: Users can install the binary package without having to compile by passing `--server='https://nvim-neorocks.github.io/rocks-binaries/'` to `luarocks install` (which is what rocks.nvim and lazy.nvim do by default).

```suggestion
      - name: Install C/C++ Compiler
        uses: rlalik/setup-cpp-compiler@master
        with:
          compiler: clang-latest

      - name: Install tree-sitter CLI
        uses: baptiste0928/cargo-install@v3
        with:
          crate: tree-sitter-cli

      - name: LuaRocks Upload
```